### DISCUSSION TITLE
BaseUnityPlugin destroyed before first Frame Update!

### DISCUSSION THREAD

**Discussion by bugerry87** [BepInEx/BepInEx#827] [Help and Q&A] [answered]
## What is Happening

Since the latest update of *Angel Legion - Idle RPG* all BepInEx BaseUnityPlugin Scripts get destroyed before the first frame update.
In consequence, no Unity Pipeline Functions can be used such as Start, Update, OnGUI, etc.
Expect, all Harmony Patches are successfully injected and working well.

## My Question

Does anyone know about a 3rd Party component that detects BepInEx Scripts and deletes them?
Or is it rather custom code by the game that either accidently or willingly destroys my mod scripts?
Are there any Unity, Steam or Windows security settings which could cause this?

## Environment

OS: Windows 11
Game: Angel Legion (Steam)
Unity: 2021.3.15f1
BepInEx: 5.4.22

## How to Replicate

Try this simple Plugin:

```
[BepInPlugin("bugerry.Tester", "Tester", "0.0.1")]
	public partial class Tester : BaseUnityPlugin
	{
		protected void Awake()
		{
			Logger.LogInfo("Awake");
		}

		protected void Start()
		{
			Logger.LogInfo("Start");
		}

		protected void Update()
		{
			Logger.LogInfo("Update");
		}

		protected void OnDestroy()
		{
			Logger.LogInfo("OnDestroy");
		}
	}
```

And receive the following BepInEx Logs (Log Level Info!):

```
[Message:   BepInEx] BepInEx 5.4.22.0 - AngelLegion (29.08.2023 13:41:47)
[Info   :   BepInEx] Running under Unity v2021.3.15.15263878
[Info   :   BepInEx] CLR runtime version: 4.0.30319.42000
[Info   :   BepInEx] Supports SRE: True
[Info   :   BepInEx] System platform: Bits64, Windows
[Message:   BepInEx] Preloader started
[Info   :   BepInEx] Loaded 1 patcher method from [BepInEx.Preloader 5.4.22.0]
[Info   :   BepInEx] 1 patcher plugin loaded
[Info   :   BepInEx] Patching [UnityEngine.CoreModule] with [BepInEx.Chainloader]
[Message:   BepInEx] Preloader finished
[Message:   BepInEx] Chainloader ready
[Message:   BepInEx] Chainloader started
[Info   :   BepInEx] 1 plugin to load
[Info   :   BepInEx] Loading [Tester 0.0.1]
[Info   :    Tester] Awake
[Message:   BepInEx] Chainloader startup complete
[Info   :    Tester] OnDestroy
```

This happens as soon the game starts. Neither Start nor Update where called. Therewith, the Tester got destroyed before the Scene Tree was build.  How could they do that? I am sure that all BepInEx Scripts are loaded to "DoNotDestroyOnLoad" such that they survive a Scene swap.

I am still checking the source code of the game for a line that is greedily destroying all object. This is most likely the issue, but I can't find such thing.

So again my Question, does anyone know about higher instances that could cause such behavior? 

PS: All instances in BepInEx.Bootstrap.Chainloader are NULL

**Comment by toebeann** [comment, marked as answer]
In the BepInEx.cfg, try setting:

```toml
[Chainloader]
HideManagerGameObject = true
```

**Reply by bugerry87** [reply]
Works!
